app/src/main/java/org/thoughtcrime/securesms/onboarding/LandingActivity.kt

```diff
@@ -19,28 +83,182 @@ class LandingActivity : BaseActionBarActivity() {
         // Session then close this activity to resume the last activity from the previous instance.
         if (!isTaskRoot) { finish(); return }
 
-        val binding = ActivityLandingBinding.inflate(layoutInflater)
-        setContentView(binding.root)
         setUpActionBarSessionLogo(true)
-        with(binding) {
-            fakeChatView.startAnimating()
-            registerButton.setOnClickListener { register() }
-            restoreButton.setOnClickListener { link() }
-            linkButton.setOnClickListener { link() }
-        }
+
+        ComposeView(this)
+            .apply { setContent { AppTheme { LandingScreen() } } }
+            .let(::setContentView)
+
         IdentityKeyUtil.generateIdentityKeyPair(this)
         TextSecurePreferences.setPasswordDisabled(this, true)
         // AC: This is a temporary workaround to trick the old code that the screen is unlocked.
         KeyCachingService.setMasterSecret(applicationContext, Object())
     }
 
-    private fun register() {
-        val intent = Intent(this, RegisterActivity::class.java)
-        push(intent)
+    @Preview
+    @Composable
+    private fun LandingScreen(
+        @PreviewParameter(ThemeResPreviewParameterProvider::class) themeResId: Int
+    ) {
+        PreviewTheme(themeResId) {
+            LandingScreen()
+        }
+    }
+
+    @Composable
+    private fun LandingScreen() {
+        var count by remember { mutableStateOf(0) }
+        val listState = rememberLazyListState()
+
+        LaunchedEffect(Unit) {
+            delay(500.milliseconds)
+            while(count < MESSAGES.size) {
+                count += 1
+                listState.animateScrollToItem(0.coerceAtLeast((count - 1)))
+                delay(1500L)
+            }
+        }
+
+        Column(modifier = Modifier.padding(horizontal = 36.dp)) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                stringResource(R.string.onboardingBubblePrivacyInYourPocket),
+                modifier = Modifier.align(Alignment.CenterHorizontally),
+                style = MaterialTheme.typography.h4,
+                textAlign = TextAlign.Center
+            )
+            Spacer(modifier = Modifier.height(24.dp))
+
+            LazyColumn(
+                state = listState,
+                modifier = Modifier
+                    .heightIn(min = 200.dp)
+                    .fillMaxWidth()
+                    .weight(2f),
```

Comment:
This layout does not display well as it stands on a large phone like a Pixel 6. I can see the top message being cut off in spite of having ample space for the list to expand into.
This is due to the forced weight factor of 2 to 1 between the list and the spacing.
The quickest way to make it work while keeping this existing structure is by adding weight to the list (3f).
What do you think? Any potential drawbacks?